hnn_core/tests/test_network.py

```diff
@@ -41,7 +41,10 @@ def test_network_models():
     add_erp_drives_to_jones_model(net_default)
     for drive_name in ['evdist1', 'evprox1', 'evprox2']:
         assert drive_name in net_default.external_drives.keys()
-    assert len(net_default.connectivity) == n_conn + 14
+    # 15 drive connections are added as follows: evdist1: 3 ampa + 3 nmda,
+    # evprox1: 4 ampa, evprox2: 4 ampa, and 1 extra zero-weighted ampa
+    # evdist1->L5_basket connection is added to comply with legacy_mode
+    assert len(net_default.connectivity) == n_conn + 15
```

Comment:
could we do legacy_mode = False in all/most of our tests unless we are explicitly testing the legacy mode ?

Reply:
Mostly yes. Aside from the tests like this one that explicitly test against a hardcoded value, I've tried to write as many of my tests as possible to accommodate either scenario. That said, it's something we should probably dedicate a whole PR to (see #393).